### PR TITLE
cmd/testbot: fetch pull request instead of branch ref

### PR DESCRIPTION
**Pull request by boymanjor**
This fixes a bug that didn't allow testbot to run
against pull requests from forked repos.

### CODE REVIEW

**Comment by kr** [comment]
:eyes:

**Comment by boymanjor** [comment]
PTAL

**Comment by boymanjor** [comment]
@kr is it fine to just fetch the pr references and use the commit hashes?

**Comment by kr** [comment]
> is it fine to just fetch the pr references and use the commit hashes?

Yeah that should work great.

**Comment by kr** [comment]
:eyes:

File: cmd/testbot/main.go [hunk 1]
```diff
@@ -35,10 +35,12 @@ var (
 type pullRequest struct {
 	Action string
 	PR     struct {
-		Head struct {
+		Number string
+		Head   struct {
 			Ref string
 			Sha string
```
Comment (kr) [inline cmd/testbot/main.go]: (nit: out of scope for this PR, but this should be named `SHA` rather than `Sha`)

**Comment by kr** [comment]
LGTM

**Pull request merged**